lib/policyEvaluator/evaluator.js

```diff
@@ -0,0 +1,286 @@
+                requestAction, policyAction: statementAction[i],
+            });
+            return true;
+        }
+    }
+    log.trace('no action in policy applicable to request action',
+        { requestAction });
+    // If no match found, return false
+    return false;
+}
+
+/**
+ * Check whether request meets policy conditions
+ * @param {Object} requestContext - info about request
+ * @param {Object} statementCondition - Condition statement from policy
+ * @param {Object} log - logger
+ * @return {boolean} true if meet conditions, false if not
+ */
+function meetConditions(requestContext, statementCondition, log) {
+    // The Condition portion of a policy is an object with different
+    // operators as keys
+    const operators = Object.keys(statementCondition);
+    const length = operators.length;
+    for (let i = 0; i < length; i ++) {
+        const operator = operators[i];
+        const hasIfExistsCondition = operator.endsWith('IfExists');
+        // If has "IfExists" added to operator name, find operator name
+        // without "IfExists"
+        const bareOperator = hasIfExistsCondition ? operator.slice(0, -8) :
+            operator;
+        const operatorsWithVariables = ['StringEquals', 'StringNotEquals',
+            'StringEqualsIgnoreCase', 'StringNotEqualsIgnoreCase',
+            'StringLike', 'StringNotLike', 'ArnEquals', 'ArnNotEquals',
+            'ArnLike', 'ArnNotLike'];
+        const operatorCanHaveVariables =
+            operatorsWithVariables.indexOf(bareOperator) > -1;
+        // Loop through conditions with the same operator
+        // Note: this should be the actual operator name, not the bareOperator
+        const conditionsWithSameOperator = statementCondition[operator];
+        for (let j = 0; j < Object.keys(conditionsWithSameOperator).length;
+            j ++) {
+            const key = Object.keys(conditionsWithSameOperator)[j];
+            let value = conditionsWithSameOperator[key];
+            if (!Array.isArray(value)) {
+                value = [value];
+            }
+            // Handle variables
+            if (operatorCanHaveVariables) {
+                value = value.map(item =>
+                    substituteVariables(item, requestContext));
+            }
+            // Pull key using requestContext
+            // TODO: If applicable to S3, handle policy set operations
+            // where a keyBasedOnRequestContext returns multiple values and
+            // condition has "ForAnyValue" or "ForAllValues".
+            // (see http://docs.aws.amazon.com/IAM/latest/UserGuide/
+            // reference_policies_multi-value-conditions.html)
+            const keyBasedOnRequestContext =
+                findConditionKey(key, requestContext);
+            // Handle IfExists
+            if (keyBasedOnRequestContext === undefined
+                && hasIfExistsCondition) {
+                log.trace('request does not contain info to satisfy condition' +
```

Comment:
We should aim for short, succinct log messages, long messages are less readable in Kibana and log files.